src/main/kotlin/org/rust/lang/core/types/BoundElement.kt

```diff
@@ -7,36 +7,38 @@ package org.rust.lang.core.types
 
 import com.intellij.psi.PsiElement
 import com.intellij.psi.ResolveResult
+import org.rust.lang.core.psi.RsTypeAlias
 import org.rust.lang.core.psi.ext.RsElement
 import org.rust.lang.core.types.infer.TypeFoldable
 import org.rust.lang.core.types.infer.TypeFolder
 import org.rust.lang.core.types.infer.TypeVisitor
 import org.rust.lang.core.types.ty.Substitution
+import org.rust.lang.core.types.ty.Ty
 import org.rust.lang.core.types.ty.emptySubstitution
 import org.rust.lang.core.types.ty.foldValues
-import org.rust.lang.core.types.ty.substituteInValues
 
-/* Represents a potentially generic Psi Element, like
+/**
+ * Represents a potentially generic Psi Element, like
  * `fn make_t<T>() -> T { ... }`, together with actual
  * type arguments, like `T := i32`.
  */
 data class BoundElement<out E : RsElement>(
     val element: E,
-    val subst: Substitution = emptySubstitution
+    /** Generic type bindings, e.g. `type T = Box<u8>;` */
+    val subst: Substitution = emptySubstitution,
+    /** Associated type bindings, e.g. `type T = Iterator<Item=u8>;` */
+    val assoc: Map<RsTypeAlias, Ty> = emptyMap()
```

Comment:
Maybe slightly change docs for these fields? It's a little bit confusing and I spent some time to understand it.
Maybe something like this:
"Generic type bindings, e.g. if we have concrete type `Box<u8>` with a generic declaration as `Box<T>`, `subst` maps `T` to `u8`"? 
And something similar for `assoc` map as well